### DISCUSSION TITLE
Newlib version header file

### DISCUSSION THREAD

**Discussion by jasr93** [xpack-dev-tools/arm-none-eabi-gcc-xpack#24] [Q&A]
Hi everyone,

I upgraded my toolchain from 11.2.1-1.2.2 to 11.3.1-1.1.2 and I have a build error related to Newlib version.

The _newlib_version.h header file provided by the toolchain has the following content:

/* _newlib_version.h.  Generated from _newlib_version.hin by configure.  */
/* Version macros for internal and downstream use. */
#ifndef _NEWLIB_VERSION_H__
#define _NEWLIB_VERSION_H__ 1

/* The newlib version in string format. */
#define _NEWLIB_VERSION "@NEWLIB_VERSION@"

/* The newlib major version number. */
#define __NEWLIB__ @NEWLIB_MAJOR_VERSION@

/* The newlib minor version number. */
#define __NEWLIB_MINOR__ @NEWLIB_MINOR_VERSION@

/* The newlib patch level. */
#define __NEWLIB_PATCHLEVEL__ @NEWLIB_PATCHLEVEL_VERSION@

#endif /* !_NEWLIB_VERSION_H__ */

However, it seems that the placeholders have not been changed to their corresponding numbers and gcc complains with "error: token "@" is not valid in preprocessor expressions"

I was able to verify that in the 11.2.1-1.2.2 toolchain, this file is formatted correctly with numbers instead of placeholders.

Thanks in advance

**Comment by ilg-ul** [comment]
This seems to be a bug in the build scripts. 

I'll investigate.

**Comment by ilg-ul** [comment]
https://github.com/xpack-dev-tools/arm-none-eabi-gcc-xpack/issues/25
